### PR TITLE
treewide: More meta pkg config modules

### PR DESCRIPTION
**Pull request by Ericson2314**
###### Description of changes

More `meta.pkgConfigModules`.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/icu/base.nix [hunk 1]
```diff
@@ -2,6 +2,7 @@
 { stdenv, lib, fetchurl, fixDarwinDylibNames
   # Cross-compiled icu4c requires a build-root of a native compile
 , buildRootOnly ? false, nativeBuildRoot
+, testers
 }:
```
Comment (roberth) [inline pkgs/development/libraries/icu/base.nix:6]: This function isn't callPackage'd, resulting in an eval error

File: pkgs/development/libraries/ffmpeg/generic.nix [hunk 2]
```diff
@@ -671,7 +677,8 @@ stdenv.mkDerivation rec {
       ++ optional withGPL gpl2Plus
       ++ optional withGPLv3 gpl3Plus
       ++ optional withUnfree unfreeRedistributable;
+    pkgConfigModules = [ "libavutil" ];
```
Comment (roberth) [inline pkgs/development/libraries/ffmpeg/generic.nix:680]: pkg-config? [no problem!](https://www.youtube.com/watch?v=9kaIXkImCAM)

Also `libavutil` - what? - and ok then.

**Review by roberth — APPROVED** [review]
Smells good.

**Comment by deviant** [comment]
At least for `jack` (I wouldn't be surprised if other packages were affected as well), this breaks the source derivation:

```
nix-repl> jack2.src.urls
[ "https://github.com/jackaudio/jack2/archive/v23.05pre-git.tar.gz" ]
```

Safe to say, the NixOS version should not be ending up in random package source URLs. I haven't the slightest clue what is causing this, either, and I'm a little afraid to look.

**Comment by winterqt** [comment]
It was because of the switch to overlay-style `mkDerivation`: https://github.com/NixOS/nixpkgs/blob/2d84e9d64687110e21b6867ac7a058155f6f9a9e/pkgs/misc/jackaudio/default.nix#L30-L39

This should be switched to `finalAttrs.version`.

**Comment by winterqt** [comment]
Haven't looked at the diff further so @Ericson2314 may have missed the others, too.

**Comment by winterqt** [comment]
It was just JACK: #217419.

**Comment by Ericson2314** [comment]
Thanks @winterqt!

**Pull request merged**